### PR TITLE
Feature/support getting from array maps

### PR DESCRIPTION
**Pull request by rv-rsouza**
Hi. 

I've stumbled on arrays of objects for some days and find out that Viper didn't supported getting parameters using index keys. This PR is an implementation proposal to add that support.

Example usage:

Config file:

```toml
[[book]]
title = "My book"
    [book.author]
    name = "Me"

[[book]]
title = "My second book"
    [book.author]
    name = "Me again"
```

We can unmarshal this structure with no problems, but there are times when we need to get some parameters that are not mapped to a struct.

```go
// ...
// initialization and config loading
// ...
var firstBookTitle = viper.GetString("book[0].title")
fmt.Println(firstBookTitle) // prints: My book

var secondBookAuthor = viper.GetString("book[1].author.name")
fmt.Println(secondBookAuthor) // prints: Me again
```

### CODE REVIEW

File: viper_test.go [hunk 1]
```diff
@@ -1400,6 +1400,39 @@ func TestParseNested(t *testing.T) {
...
+`
+	initConfig("toml", config)
+
+	firstNestedDelay := v.GetDuration("parent[0].delay")
+
+	assert.Equal(t, 100*time.Millisecond, firstNestedDelay)
+
+	secondNestedDelay := v.GetDuration("parent[1].nested.delay")
+
+	assert.Equal(t, 400*time.Millisecond, secondNestedDelay)
```
[the remark this answers is not in the document]
Comment (rv-rsouza) [reply viper_test.go]: Working as expected. What do you think?

**Comment by rv-rsouza** [comment]
Hey guys! I'm using this for a while now in production for more than 6 months, using the forked code, and had no issues yet.
Do I need to do anything more to get this merged?

File: viper_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rcdmk) [reply viper_test.go]: Hi. Is there anything that needs to be done here? It seems to be working fine and the test cases were added as requested.

**Comment by rv-rsouza** [comment]
Closing this as I had no feedback for a while and the need for this has gone in favor of using a different structure to unmarshal. (eg: Using maps instead of slices for struct collections).

**Pull request closed without merging**